Add note about `log_fmt` in esp32 compact logs (#321)
memfault/memfault-docs: docs/mcu/compact-logs.mdx

const char *str = "hello world";
MEMFAULT_TRACE_EVENT_WITH_LOG("String Pointer: %p", (void *)str);
```

### For ESP-IDF builds, which symbol file should I upload?

For ESP32 builds, Memfault suggests including some additional `POST_BUILD`
commands to strip the `log_fmt` section from the executable. These commands can
be added to the application's top-level `CMakeLists.txt`:

> ```cmake

Author: done!

```

This unfortunately causes the `esptool.py` to output the `log_fmt` section into
the flashable `.bin` that's generated as part of the build:

```bash
❯ ../../esp-idf/components/esptool_py/esptool/esptool.py --chip esp32 image_info build/memfault-esp32-demo-app.bin
esptool.py v3.1-dev
Image version: 1
Entry point: 40081a98
7 segments

Segment 1: len 0x25174 load 0x3f400020 file_offs 0x00000018 [DROM]
Segment 2: len 0x03b30 load 0x3ffb0000 file_offs 0x00025194 [BYTE_ACCESSIBLE,DRAM]
Segment 3: len 0x07344 load 0x40080000 file_offs 0x00028ccc [IRAM]
Segment 4: len 0xa49d8 load 0x400d0020 file_offs 0x00030018 [IROM]
Segment 5: len 0x106a4 load 0x40087344 file_offs 0x000d49f8 [IRAM]
Segment 6: len 0x00064 load 0x400c0000 file_offs 0x000e50a4 [RTC_IRAM]
# see this bad section below! this will cause a bootloop on the board
Segment 7: len 0x000a8 load 0xf0000000 file_offs 0x000e5110 []
Checksum: 7f (valid)
Validation Hash: fd388c619d75da39d9646b247bdf568a69a6e2c5b801c88fe7bd07b300b184fb (valid)
```

Flashing that image to an ESP32 board results in a bootloop:

```plaintext
E (419) esp_image: Segment 6 0xf0000000-0xf00000a8 invalid: bad load address range
```

> The fundamental problem is in the `esptool.py` tool, which processes the `.elf`
> and produces the `.bin` for flashing (or OTA)- it includes the non-loadable
> `log_fmt` section in the output.

Reviewer: ```suggestion
The fundamental problem is that the `esptool.py` tool, which processes the .elf tool, includes the non-loadable
`log_fmt` section in the `.bin` file used for flashing or OTA.
```

const char *str = "hello world";
MEMFAULT_TRACE_EVENT_WITH_LOG("String Pointer: %p", (void *)str);
```

### For ESP-IDF builds, which symbol file should I upload?

> For [ESP-IDF](/docs/mcu/esp32-guide), Memfault recommends using the

Author: This is way too much background information + detail I think. I might move it behind a "More details on why this is necessary" toggle instead :confused:
